hasher called w/ UUID objects

Noticed by mypy:

autopush/web/registration.py:320: error: Argument 1 to "hasher" has incompatible type "UUID"; expected "str"

(one in _write_endpoint and _delete_channel)

This can cause hasher to blow up but only with a key_hash configured (which isn't configured on prod)

--- a/autopush/web/base.py
+++ b/autopush/web/base.py
@@ -6,7 +6,11 @@ from attr import attrs, attrib
 from boto.dynamodb2.exceptions import ProvisionedThroughputExceededException
 from boto.exception import BotoServerError
 from marshmallow.schema import UnmarshalResult  # noqa
-from typing import Any, Callable  # noqa
+from typing import (  # noqa
+    Any,
+    Callable,
+    Sequence
+)
 from twisted.internet.threads import deferToThread
 from twisted.logger import Logger
 
@@ -142,8 +146,8 @@ class Notification(object):
 class BaseWebHandler(BaseHandler):
     """Common overrides for Push web API's"""
     cors_methods = ""
-    cors_request_headers = ()
-    cors_response_headers = ()
+    cors_request_headers = ()   # type: Sequence[str]
+    cors_response_headers = ()  # type: Sequence[str]
 
     #############################################################
     #                    Cyclone API Methods

--- a/autopush/web/registration.py
+++ b/autopush/web/registration.py
@@ -41,11 +41,9 @@ class PathUUID(fields.Field):
     throw a 404"""
     def _deserialize(self, value, attr, data):
         try:
-            new_uuid = uuid.UUID(value)
+            return uuid.UUID(value)
         except ValueError:
             raise InvalidRequest("Invalid Path", status_code=404)
-        else:
-            return new_uuid
 
 
 #############################################################
@@ -317,7 +315,7 @@ class BaseRegistrationHandler(BaseWebHandler):
         self.log.info("Register",
                       client_info=self._client_info,
                       endpoint=endpoint,
-                      uaid_hash=hasher(uaid))
+                      uaid_hash=hasher(uaid.hex))
         self.finish()
 
     def _success(self, result):
@@ -355,7 +353,7 @@ class NewRegistrationHandler(BaseRegistrationHandler):
         return d
 
     def _register_user_and_channel(self, uaid, chid, router_type, router_data):
-        # type: (uuid.UUID, str, str, JSONDict, Optional[str]) -> str
+        # type: (uuid.UUID, str, str, JSONDict) -> str
         """Register a new user/channel, return its endpoint"""
         self._register_user(uaid, router_type, router_data)
         return self._register_channel(uaid, chid, router_data.get("key"))
@@ -455,7 +453,7 @@ class ChannelRegistrationHandler(BaseRegistrationHandler):
 
     @threaded_validate(UnregisterChidSchema)
     def delete(self, uaid, chid):
-        # type: (uuid.UUID, str) -> Deferred
+        # type: (uuid.UUID, uuid.UUID) -> Deferred
         self.metrics.increment("ua.command.unregister",
                                tags=self.base_tags())
         d = deferToThread(self._delete_channel, uaid, chid)
@@ -465,12 +463,13 @@ class ChannelRegistrationHandler(BaseRegistrationHandler):
         return d
 
     def _delete_channel(self, uaid, chid):
-        if not self.db.message.unregister_channel(uaid.hex, chid):
+        # type: (uuid.UUID, uuid.UUID) -> None
+        if not self.db.message.unregister_channel(uaid.hex, chid.hex):
             raise ItemNotFound("ChannelID not found")
         self.log.info("Unregister",
                       client_info=self._client_info,
-                      channel_id=chid,
-                      uaid_hash=hasher(uaid))
+                      channel_id=chid.hex,
+                      uaid_hash=hasher(uaid.hex))
 
     def _chid_not_found_err(self, fail):
         """errBack for unknown chid"""

--- a/autopush/web/webpush.py
+++ b/autopush/web/webpush.py
@@ -34,6 +34,7 @@ from autopush.exceptions import (
     InvalidTokenException,
     VapidAuthException,
 )
+from autopush.types import JSONDict  # noqa
 from autopush.utils import (
     base64url_encode,
     extract_jwt,
@@ -459,7 +460,7 @@ class WebPushHandler(BaseWebHandler):
     def post(self,
              subscription,  # type: Dict[str, Any]
              notification,  # type: WebPushNotification
-             jwt=None,      # type: Optional[Dict[str, str]]
+             jwt=None,      # type: Optional[JSONDict]
              **kwargs       # type: Any
              ):
         # type: (...) -> Deferred

